### PR TITLE
Dusting things off

### PR DESCRIPTION
**Pull request by DFurnes**
#### Changes

CSS, what's that?! :information_desk_person: This PR includes some "housekeeping" that I wanted to bunch up with our Forge 6.7 changes happening this sprint.

:100: Updates ESLint rules & uses a shared config package [`@dosomething/eslint-config`](https://github.com/DoSomething/eslint-config) that I'd made back when we did our documentation sprint. Makes some tiny updates in cases where we were following slightly different rules before.

:construction_worker: Similarly, extracts Webpack config into a shared [`@dosomething/webpack-config`](https://github.com/DoSomething/webpack-config) package that we can share between all of our projects (inspired by [Fender](https://github.com/DFurnes/fender)!)

:art: And with that new shared Webpack config, we can also use Webpack to build our stylesheets! This lets Webpack handle asset bundling/hashing & opens us up to some other cool possibilities in the future.

:speech_balloon: Updates Babel, our JavaScript compiler, to 6.x.

:chart_with_upwards_trend: Updates to Modernizr v3. This removes the (often buggy) `grunt-modernizr` plugin in favor of just using the official CLI directly. Good news: we can now use the bundled `checked` test rather than our copied version. Bad news: Modernizr no longer supports including custom tests, so we just have to "fake" a test in our `forge.js` bundle for the "label click" test.

:wrench: Splits out Susy & helpers into a `toolkit.scss` file. Also bundles up our "base" styles into a `base.scss` that could be included when making projects that don't use the full DS branding, like Voting App.

:door: Adds a new `index.js` entry point in the root of the project. This acts as a table-of-contents for loading in styles and scripts, and sets the stage for potentially having multiple entry points in future releases (e.g. a `forge/admin` or `forge/lite`, etc.)

:whale: Updates Wercker config to use their new Docker stack. We needed a more up-to-date version of NPM than was included on the classic Node stack, so why not!

---

For review: @DoSomething/front-end

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,4 +23,4 @@ You can also run the pattern library on your local machine! Here's the steps:
   4. Check it out at `localhost:3000` in your browser!
 
 ### License
-&copy;2015 DoSomething.org. Forge is free software, and may be redistributed under the terms specified in the [LICENSE](https://github.com/DoSomething/forge/blob/dev/LICENSE) file. The name and logo for DoSomething.org are trademarks of Do Something, Inc and may not be used without permission.
+&copy;2016 DoSomething.org. Forge is free software, and may be redistributed under the terms specified in the [LICENSE](https://github.com/DoSomething/forge/blob/dev/LICENSE) file. The name and logo for DoSomething.org are trademarks of Do Something, Inc and may not be used without permission.
```
Comment (sbsmith86) [inline README.md:26]: New year! 🎉🎉

**Comment by weerd** [comment]
awesome-sauce :spaghetti:  on pulling out the configs into separate repos :dancers:

**Comment by weerd** [comment]
> Updates to Modernizr v3. This removes the (often buggy) grunt-modernizr plugin in favor of just using the official CLI directly.

yaaaais!

**Comment by weerd** [comment]
> Bad news: Modernizr no longer supports including custom tests, so we just have to "fake" a test in our forge.js bundle for the "label click" test.

Really? that seems like a step backwards for them :(

**Comment by DFurnes** [comment]
Yeah, it's a bit of a bummer. There's an [open issue](https://git.io/vadFQ) discussing it, but it seems like the fix is blocked due to potential security concerns because the web interface uses the same "builder" codebase. :confused:

**Comment by weerd** [comment]
Cool. So it seems like they want to add it, but the security concern is the hold up. Once they figure something out we can always do an upgrade if we need to add custom stuff in :)

File: Gruntfile.js [hunk 2]
```diff
@@ -245,11 +197,11 @@ module.exports = function(grunt) {
 
   // > grunt build
   // Build for production.
-  grunt.registerTask('build', ['clean:dist', 'copy:assets', 'sass:prod', 'postcss:prod', 'webpack:prod', 'modernizr:all']);
+  grunt.registerTask('build', ['clean:dist', 'copy:assets', 'sass:prod', 'postcss:prod', 'webpack', 'modernizr:all']);
 
   // > grunt build:debug
   // Build for development.
-  grunt.registerTask('build:debug', ['clean:dist', 'copy:assets', 'sass:debug', 'postcss:debug', 'webpack:debug', 'modernizr:all']);
+  grunt.registerTask('build:debug', ['clean:dist', 'copy:assets', 'sass:debug', 'postcss:debug', 'webpack', 'modernizr:all']);
```
Comment (weerd) [inline Gruntfile.js]: So `webpack` task is the same for both prod and debug?

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  },
+  module: {
+    loaders: [
+      { test: /\.js$/, exclude: /node_modules/, loader: 'babel-loader'}
+    ]
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      'process.env': {
+        'NODE_ENV': process.env.NODE_ENV
```
Comment (weerd) [inline webpack.config.js]: ? is `NODE_ENV` a constant setup on each environment?

File: .babelrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015", "react"]
```
Comment (weerd) [inline .babelrc]: image

File: Gruntfile.js (hunk 2 above)
Comment (DFurnes) [inline Gruntfile.js]: Yep, the environment switching is handled by `NODE_ENV` in the Webpack config. This was a bit of a stop-gap on the way to completely pulling this stuff out.

File: webpack.config.js (hunk 3 above)
Comment (DFurnes) [inline webpack.config.js]: It's a pretty standard way of setting debug vs. production environments in the Node world. React, for example, uses this to conditionally switch on and off debugging features in their builds so it seems to make sense for us to do the same. :)

**Comment by weerd** [comment]
hooooboy! lots of stuff, but all really :sunglasses: 

:+1:

**Comment by DFurnes** [comment]
:bow:

**Pull request merged**